### PR TITLE
Use canonical JSON for all metadata when signing

### PR DESCRIPTION
**Pull request by karlb**
As noticed in https://github.com/raiden-network/raiden/issues/7052,
using canonical_json to encode the address_metadata breaks backwards
compatibility. When we break compatibility, let's do the right way and
use canonical_json for all of the metadata. Advantages:
* No mix of RLP and JSON encoding
* Can support dicts in all parts of the metadata
* Same encoding can be used when passing along unknown metadata.
Previously, we would have needed to agree on how to encode every new
piece of metadata.

Fixes https://github.com/raiden-network/raiden/issues/7052, as soon as the analogous change has been implemented in the LC.
Replaces and thus closes #7070

### CODE REVIEW

File: raiden/messages/metadata.py [hunk 1]
```diff
@@ -31,22 +30,22 @@ def _validate_address_metadata(self) -> None:
...
+        if not self.address_metadata:
+            # We add a default value of {} when validating. Normalize this to
+            # no key when signing.
+            return {"route": route}
+
+        address_metadata = {
+            to_checksum_address(address): metadata
+            for address, metadata in self.address_metadata.items()
+        }
+        return {"route": route, "address_metadata": address_metadata}
```
Comment (andrevmatos) [inline raiden/messages/metadata.py:48]: Isn't this hardcoding the keys allowed on this schema? Can this support arbitrary unknown fields?

Comment (karlb) [reply raiden/messages/metadata.py:48]: The current implementation can't deal with unknown fields, yet. But we can add support for unknown fields without changing the encoding yet again. I wanted to keep this change as small as possible and add support for actually handling arbitrary data in a later PR.

**Comment by fredo** [comment]
This PR makes your previous one #7070 obsolete right? Or rather it also includes the proper change?

**Comment by karlb** [comment]
Yes, this is an alternative to #7070, if we can agree to go this way.

**Comment by fredo** [comment]
Ok then I'll wait for the outcome of the LC call

**Comment by andrevmatos** [comment]
We've agreed we can do the canonical metadata signing. We'll be pushing it to our `next` branch. But we'd still prefer if `address_metadata` was fully optional, and the fallback of fetching it from PFS if needed was implemented.

**Review by fredo — APPROVED** [review]
Let's go for it now, and address the pending issues in later PRs

**Comment by andrevmatos** [comment]
We've implemented it as part of our `next` branch.

**Pull request merged**